src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,128 @@
+            if (resultSet.next()) {
+                Book createdBook = createBook(resultSet);
+                return Optional.of(createdBook);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id. ID = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                books.add(createBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books.", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(FIRST_INDEX_PARAMETER, book.getTitle());
+            statement.setBigDecimal(SECOND_INDEX_PARAMETER, book.getPrice());
+            statement.setLong(THIRD_INDEX_PARAMETER, book.getId());
+            int affectedRows = statement.executeUpdate();
+            checkInsertedRowsAmount(affectedRows);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update the book: " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        String query = "DELETE FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(FIRST_INDEX_PARAMETER, id);
+            int affectedRows = statement.executeUpdate();
+            return affectedRows > ZERO_ROWS;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete book by id. ID = " + id, e);
+        }
+    }
+
+    private void checkInsertedRowsAmount(int affectedRows) {
+        if (affectedRows < ONE_ROW) {
+            throw new RuntimeException("Expected to insert at least 1 row, but inserted 0 rows.");
+        }
+    }
+
+    private Book createBook(ResultSet resultSet) throws SQLException {
+        Book book = new Book();
+        book.setId(resultSet.getObject("id", Long.class));
+        book.setTitle(resultSet.getString("title"));
+        book.setPrice(getBookPrice(resultSet));
+        return book;
+    }
+
+    private BigDecimal getBookPrice(ResultSet resultSet) throws SQLException {
+        return resultSet.getBigDecimal("price") != null
```

Comment:
It is not require to create new method to hadle it, just do tha same as you did for id